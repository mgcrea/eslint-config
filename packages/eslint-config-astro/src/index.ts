/** @type {import('eslint').Linter.Config} */
module.exports = {
  extends: ["@mgcrea/eslint-config-react", "plugin:astro/recommended"],
  overrides: [
    {
      files: ["*.astro"],
      parser: "astro-eslint-parser",
      parserOptions: {
        parser: "@typescript-eslint/parser",
        extraFileExtensions: [".astro"],
      },
      rules: {},
    },
  ],
  rules: {},
};
