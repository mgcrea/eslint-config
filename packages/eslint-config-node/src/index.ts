/* eslint-disable no-undef */

/** @type {import('eslint').Linter.Config} */
module.exports = {
  extends: [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:prettier/recommended",
  ],
  parser: "@typescript-eslint/parser",
  parserOptions: {
    warnOnUnsupportedTypeScriptVersion: false,
  },
  plugins: ["@typescript-eslint"],
  settings: {
    react: {
      version: "detect",
    },
  },
  overrides: [
    {
      files: ["*.{spec,test}.{js,ts,tsx}", "**/__{mocks,tests}__/**/*.{js,ts,tsx}"],
      extends: ["plugin:vitest/recommended"],
      rules: {
        "vitest/no-focused-tests": ["error"],
      },
    },
  ],
  rules: {
    "@typescript-eslint/no-unused-vars": [
      "warn",
      { ignoreRestSiblings: true, argsIgnorePattern: "^_" },
    ],
  },
};
