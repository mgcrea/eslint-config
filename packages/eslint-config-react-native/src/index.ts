import { fixupPluginRules, type FlatConfig } from "@eslint/compat";
import baseConfig from "@mgcrea/eslint-config-react";
import reactNative from "eslint-plugin-react-native";
import { globals } from "./globals";

const config = [
  ...baseConfig,
  {
    files: ["*.{js,jsx,ts,tsx}"],
    plugins: {
      "react-native": fixupPluginRules(reactNative),
    },
    languageOptions: {
      parserOptions: { ecmaFeatures: { jsx: true } },
      globals: { ...globals },
    },
  },
];

export default config;
