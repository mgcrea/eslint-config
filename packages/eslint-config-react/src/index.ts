import { fixupPluginRules } from "@eslint/compat";
import eslint from "@eslint/js";
import vitest from "@vitest/eslint-plugin";
import prettierRecommended from "eslint-plugin-prettier/recommended";
import react from "eslint-plugin-react";
import reactHooks from "eslint-plugin-react-hooks";
import reactRefresh from "eslint-plugin-react-refresh";
import testingLibrary from "eslint-plugin-testing-library";
import globals from "globals";
import tseslint, { type Config } from "typescript-eslint";

type NoPromise<T> = T extends Promise<unknown> ? never : T;

const config: NoPromise<Config> = tseslint.config(
  eslint.configs.recommended,
  ...tseslint.configs.strictTypeChecked,
  ...tseslint.configs.stylisticTypeChecked,
  prettierRecommended,
  {
    rules: {
      "@typescript-eslint/no-unused-vars": [
        "warn",
        { ignoreRestSiblings: true, argsIgnorePattern: "^_", varsIgnorePattern: "^_" },
      ],
      "@typescript-eslint/consistent-type-definitions": ["warn", "type"],
      "@typescript-eslint/no-unnecessary-condition": ["warn"],
      "@typescript-eslint/restrict-template-expressions": ["warn", { allowNumber: true }],
      "@typescript-eslint/ban-ts-comment": ["error", { minimumDescriptionLength: 3 }],
    },
  },
  {
    files: ["*.{js,jsx,ts,tsx}"],
    plugins: {
      react,
      "react-hooks": reactHooks,
      "react-refresh": fixupPluginRules(reactRefresh),
    },
    languageOptions: {
      parserOptions: { ecmaFeatures: { jsx: true } },
      globals: { ...globals.browser },
    },
    rules: {
      ...react.configs.recommended.rules,
      ...reactHooks.configs.recommended.rules,
    },
  },
  {
    files: ["*.{mock,spec,test}.{js,ts,tsx}", "**/__{mocks,tests}__/**/*.{js,ts,tsx}"],
    plugins: {
      vitest,
      "testing-library": fixupPluginRules(testingLibrary),
    },
    rules: {
      ...testingLibrary.configs["flat/react"].rules,
      ...vitest.configs.recommended.rules,
      "@typescript-eslint/no-unsafe-assignment": "off",
    },
  },
);

export default config;
