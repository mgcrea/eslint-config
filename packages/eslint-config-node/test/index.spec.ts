import { describe, expect, it } from "vitest";
import * as module from "./../src";

describe("module", () => {
  it("should export an eslint config", () => {
    expect(module).toMatchSnapshot();
  });
});
