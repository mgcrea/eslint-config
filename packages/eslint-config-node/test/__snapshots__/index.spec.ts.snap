// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`module > should export an eslint config 1`] = `
{
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:prettier/recommended",
  ],
  "overrides": [
    {
      "extends": [
        "plugin:vitest/recommended",
      ],
      "files": [
        "*.{spec,test}.{js,ts,tsx}",
        "**/__{mocks,tests}__/**/*.{js,ts,tsx}",
      ],
      "rules": {
        "vitest/no-focused-tests": [
          "error",
        ],
      },
    },
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "warnOnUnsupportedTypeScriptVersion": false,
  },
  "plugins": [
    "@typescript-eslint",
  ],
  "rules": {
    "@typescript-eslint/no-unused-vars": [
      "warn",
      {
        "argsIgnorePattern": "^_",
        "ignoreRestSiblings": true,
        "varsIgnorePattern": "^_",
      },
    ],
  },
  "settings": {
    "react": {
      "version": "detect",
    },
  },
}
`;
